pkg/ipcmd/ipcmd.go

```diff
@@ -0,0 +1,98 @@
+package ipcmd
+
+import (
+	"regexp"
+
+	"github.com/openshift/openshift-sdn/pkg/exec"
+)
+
+var ipcmdPath string
+
+func init() {
+	var err error
+
+	ipcmdPath, err = exec.LookPath("ip")
+	if err != nil {
+		panic("ip is not installed")
+	}
+}
+
+type Transaction struct {
+	link string
+	err  error
```

Comment:
Comment here will be helpful: Any commands on the link will not be executed if it finds existing error on the link.